src/Components/Facility/ConsultationDetails/ConsultationFeedTab.tsx

```diff
@@ -65,11 +67,23 @@ export const ConsultationFeedTab = (props: ConsultationTabProps) => {
           obj.meta.type !== "boundary",
       );
 
-      const lastPresetId = sessionStorage.getItem(
-        getFeedPresetKey(props.consultationId),
-      );
+      const lastStateJSON = sessionStorage.getItem(feedStateSessionKey);
       const preset =
-        presets.find((obj) => obj.id === lastPresetId) ?? presets[0];
+        (() => {
+          if (lastStateJSON) {
+            const lastState = JSON.parse(lastStateJSON) as LastFeedState;
+            if (lastState.type === "preset") {
+              return presets.find((obj) => obj.id === lastState.value);
+            }
+            if (lastState.type === "position") {
+              return {
+                ...presets[0],
+                id: "",
+                meta: { ...presets[0].meta, position: lastState.value },
+              };
+            }
+          }
+        })() ?? presets[0];
```

Comment:
lets use a useMemo here

Reply:
this is part of the `onResponse` of the parent `useQuery` and hence gets triggered only when the preset fetch request responds, and not on re-renders.

Reply:
<img width="662" alt="image" src="https://github.com/user-attachments/assets/9bb7eb72-40dc-401f-985f-e733741211ca">